pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1297,6 +1321,11 @@ func (lbaas *LbaasV2) checkService(service *corev1.Service, nodes []*corev1.Node
 	svcConf.keepClientIP = keepClientIP
 	svcConf.enableProxyProtocol = useProxyProtocol
 
+	svcConf.timeoutClientData = getIntFromServiceAnnotation(service, ServiceAnnotationLoadBalancerTimeoutClientData, 0)
```

Comment:
Where is the default value 0 coming from? Please check the default values from Octavia API doc.

Reply:
Isn't it just exactly the same as before? In the old code l.1126 the annotations were read if set. If they were not set the go default (which is 0) was used?

Reply:
Thank you very much for this hint. From the [Octavia API docs](https://docs.openstack.org/api-ref/load-balancer/v2/?expanded=create-listener-detail#create-listener).

Name | In | Type | Description
-- | -- | -- | --
timeout_client_data (Optional) | body | integer | Frontend client inactivity timeout in milliseconds. Default: **50000**. New in version 2.1
timeout_member_connect (Optional) | body | integer | Backend member connection timeout in milliseconds. Default: **5000**. New in version 2.1
timeout_member_data (Optional) | body | integer | Backend member inactivity timeout in milliseconds. Default: **50000**. New in version 2.1
timeout_tcp_inspect (Optional) | body | integer | Time, in milliseconds, to wait for additional TCP packets for content inspection. Default: **0**. New in version 2.1

Will change the defaults accordingly.

Reply:
Ah I think I got it wrong. In the old version the values were only propagated to Openstack when the annotations are set. Now we set them in any case so it makes sense to set the defaults